### PR TITLE
CMR-4653 Add DataFormat and MimeType to both RelatedURL GET SERVICE and GET DATA

### PR DESCRIPTION
**Pull request by daniel-zamora**

### CODE REVIEW

File: umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/dif10.clj [hunk 1]
```diff
@@ -185,7 +185,8 @@
           :let [[type subtype] (dif-util/umm-url-type->dif-umm-content-type
                                 (util/remove-nil-keys
                                  (select-keys related-url [:URLContentType :Type :Subtype])))
-                mime-type (get-in related-url [:GetService :MimeType])
+                mime-type (or (get-in related-url [:GetService :MimeType])
```
Comment (chris-durbin) [inline umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/dif10.clj]: Is there any reason to prioritize the MimeType of the GetService URL over the GetData URL (or vice versa)?

File: umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/iso_shared/distributions_related_url.clj [hunk 2]
```diff
@@ -84,12 +91,66 @@
            {:codeList (str (:ngdc iso/code-lists) "#CI_OnLineFunctionCode")
             :codeListValue code}]]]])))
 
+(defn generate-online-resource-url2
```
Comment (chris-durbin) [inline umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/iso_shared/distributions_related_url.clj]: Is there also a `generate-online-resource-url` function? If so it's probably worth mentioning in this docstring why there are two and how to choose which one to use.

File: umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/distributions_related_url.clj [hunk 3]
```diff
@@ -98,37 +112,59 @@
...
 
+(defn- parse-distributor-format
+  "Parses distributor format string, returns MimeTyp and Format"
+  [distributor distributor-xpaths-map]
+  (when-let [format-name (value-of distributor (get distributor-xpaths-map :Format))]
+    (let [mime-type-index (get-index-or-nil format-name "MimeType:")
+          format-index (get-index-or-nil format-name "Format:")]
+      {:Format (when format-index
+                 (let [format (subs format-name
+                                          format-index
```
Comment (chris-durbin) [inline umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/distributions_related_url.clj]: Formatting if off. Is it ironic that the word 'format' is used 4 times in this binding?

File: umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/dif10.clj (hunk 1 above)
Comment (daniel-zamora) [reply umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/dif10.clj]: I don't think so, unless @eereiter has different ideas.

Comment (eereiter) [reply umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/dif10.clj]: no

File: umm-spec-lib/src/cmr/umm_spec/test/expected_conversion.clj [hunk 4]
```diff
@@ -8,6 +8,7 @@
     [clojure.string :as str]
```
Comment (oubiwann) [inline umm-spec-lib/src/cmr/umm_spec/test/expected_conversion.clj]: Hey, while you're in here, could you un-indent these requires by one space (to match our convention).

File: umm-spec-lib/src/cmr/umm_spec/test/expected_conversion.clj [hunk 5]
```diff
@@ -8,6 +8,7 @@
...
     [cmr.common.util :as util :refer [update-in-each]]
     [cmr.umm-spec.json-schema :as js]
+    [cmr.umm-spec.migration.version.core :as core]
```
Comment (oubiwann) [inline umm-spec-lib/src/cmr/umm_spec/test/expected_conversion.clj]: Hey, while you're in here, could you un-indent these requires by one space (to match our convention).

File: umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj [hunk 6]
```diff
@@ -239,4 +239,6 @@
 (defmethod interface/migrate-umm-version [:collection "1.10" "1.9"]
   [context c & _]
   (-> c
-      coll-progress-migration/migrate-down))
+      coll-progress-migration/migrate-down
+      (util/update-in-all [:RelatedUrls :GetData] dissoc :MimeType)
+      (util/update-in-all [:RelatedUrls :GetService] dissoc :Format)))
```
Comment (sxu123) [inline umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj]: There is umm_spec/test/migration/version/collection.clj used to test the migration. I didn't see it.

File: system-int-test/test/cmr/system_int_test/ingest/collection_ingest_test.clj [hunk 7]
```diff
@@ -607,7 +607,7 @@
        :iso-smap ["Line 1 - cvc-elt.1: Cannot find the declaration of element 'XXXX'."]))
 
 (deftest ingest-umm-json
-  (let [json (umm-spec/generate-metadata test-context exc/example-collection-record :umm-json)
+  (let [json (umm-spec/generate-metadata test-context exc/curr-ingest-ver-example-collection-record :umm-json)
```
Comment (sxu123) [inline system-int-test/test/cmr/system_int_test/ingest/collection_ingest_test.clj]: 1. The name of the record: call it current will become obsolete soon once 1.10 becomes current.
2. Seems that we want to use the record that contains the new fields to do the ingest and other testings.  Not sure why it's specifically using the version 1.9 data record and now it's hardcoded to 1.9 version and we will always use this version of test data?
3. Also, do we have any example data files that contain the newly added fields in GetService and GetData? If not, we need to add one, to make the round-trip testing meaningful.

Comment (daniel-zamora) [reply system-int-test/test/cmr/system_int_test/ingest/collection_ingest_test.clj]: Done, think I got them all.

File: umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj (hunk 6 above)
Comment (daniel-zamora) [reply umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj]: Done.

**Pull request merged**